src/helpers/figures/charts/chart_ui_common.ts

```diff
@@ -50,6 +50,33 @@ export function filterEmptyDataPoints(
   };
 }
 
+export function aggregateDataForLabels(
+  labels: string[],
+  datasets: DatasetValues[]
+): { labels: string[]; dataSetsValues: DatasetValues[] } {
+  const parseNumber = (value) => (typeof value === "number" ? value : 0);
+
+  const labelMap = {};
+  for (let indexOfLabel = 0; indexOfLabel < labels.length; ++indexOfLabel) {
```

Comment:
```suggestion
  for (const indexOfLabel of range(0, labels.length)) {
```
Slightly easier to read, more declarative, less moving parts.
`range` is inspired by the python `range` function :wink: